src/main/java/org/apache/datasketches/kll/KllDoublesSketchSortedView.java

```diff
@@ -39,56 +41,91 @@ public final class KllDoublesSketchSortedView implements DoublesSortedView {
   private final double[] quantiles;
   private final long[] cumWeights; //comes in as individual weights, converted to cumulative natural weights
   private final long totalN;
+  private final double[] normRanks;
+  private final double maxItem;
+  private final double minItem;
 
   /**
    * Construct from elements for testing.
    * @param quantiles sorted array of quantiles
    * @param cumWeights sorted, monotonically increasing cumulative weights.
    * @param totalN the total number of items presented to the sketch.
    */
-  KllDoublesSketchSortedView(final double[] quantiles, final long[] cumWeights, final long totalN) {
+  KllDoublesSketchSortedView(final double[] quantiles, final long[] cumWeights, final long totalN,
+      final double maxItem, final double minItem) {
     this.quantiles = quantiles;
     this.cumWeights  = cumWeights;
     this.totalN = totalN;
+    this.maxItem = maxItem;
+    this.minItem = minItem;
+    final int len = cumWeights.length;
+    final double[] normRanks = new double[len];
+    for (int i = 0; i < len; i++) { normRanks[i] = (double)cumWeights[i] / totalN; }
+    this.normRanks = normRanks;
   }
 
   /**
    * Constructs this Sorted View given the sketch
-   * @param sk the given KllDoublesSketch.
+   * @param sketch the given KllDoublesSketch.
    */
-  public KllDoublesSketchSortedView(final KllDoublesSketch sk) {
-    this.totalN = sk.getN();
-    final double[] srcQuantiles = sk.getDoubleItemsArray();
-    final int[] srcLevels = sk.levelsArr;
-    final int srcNumLevels = sk.getNumLevels();
+  public KllDoublesSketchSortedView(final KllDoublesSketch sketch) {
+    if (sketch.isEmpty()) { throw new SketchesArgumentException(EMPTY_MSG); }
+    this.totalN = sketch.getN();
+    this.maxItem = sketch.getMaxItem();
+    this.minItem = sketch.getMinItem();
+    final double[] srcQuantiles = sketch.getDoubleItemsArray();
+    final int[] srcLevels = sketch.levelsArr;
+    final int srcNumLevels = sketch.getNumLevels();
 
-    if (!sk.isLevelZeroSorted()) {
+    if (!sketch.isLevelZeroSorted()) {
```

Comment:
Not so simple.  The only place we care about level0 being sorted is in the sorted view.  This piece of code you are highlighting is actually in the method that creates the sorted view from the sketch.  This method happens to be in the sorted view class, but really should be associated with the sketch code itself (perhaps in its own class), which would clean up the code of the SV and make it more stand-alone. And it would allow us to reduce a bunch of code duplication across all the quantile sketches because we wouldn't need a dedicated SV for every q-sketch (This will be a code cleanup for later).

Reply:
Perhaps I did not make myself clear. My point was that it would be better to ask the sketch to do its own sorting rather than modifying it from the outside. This is knows as "feature envy code smell".

Reply:
I understood exactly what you were referring to.  And what I said above is that the method where this occurs _should_ be part of the sketch, but is not a cleanup I wanted to do now.  This sorting of level 0 has been there in this method, in this class, for a number of releases.   The only change I made in this iteration was to change the name of the variable from "sk" to "sketch" to be consistent with similar code in the other KLL SVs.  I could do this cleanup now, but it occurs in all of the SV classes and will delay this release, which I am hoping to get out before the Christmas break.

Reply:
I see, that is right, it was not introduced now.